src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1946,17 +1947,60 @@ private static bool CanReplaceWithReducedNameInContext(this NameSyntax name, Typ
             var invalidTransformation2 = WillConflictWithExistingLocal(name, reducedName);
             var invalidTransformation3 = IsAmbiguousCast(name, reducedName);
             var invalidTransformation4 = IsNullableTypeInPointerExpression(name, reducedName);
-            var isNotNullableReplaceable = name.IsNotNullableReplaceable(reducedName);
+            var invalidTransformation5 = name.IsNotNullableReplaceable(reducedName);
+            var invalidTransformation6 = IsQualifiedNameInUsingDirective(semanticModel, name, reducedName);
 
-            if (invalidTransformation1 || invalidTransformation2 || invalidTransformation3 || invalidTransformation4
-                || isNotNullableReplaceable)
+            if (invalidTransformation1 ||
```

Comment:
Can we just inline this? Each check is small enough, and then we can short circuit too.